### PR TITLE
[improve](scan)scanner could eos early when reached limit

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes

two point have change: 
1.  iff should_run_serial, it's could set _max_thread_num = 1
```
    // For select * from table limit 10; should just use one thread.
    if (_local_state && _local_state->should_run_serial()) {
        _max_thread_num = 1;
    }
```

2. scan operator have two limit, one is _limit, another is _limit_per_scanner
    could set eos when scan rows reached limit, it should use _limit variables.
```
    // If sort info is set, push limit to each scanner;
    int64_t _limit_per_scanner = -1;
    if (_olap_scan_node.__isset.sort_info && _olap_scan_node.__isset.sort_limit) {
        _limit_per_scanner = _olap_scan_node.sort_limit;
    }
```

```
before and after:
mysql [ssb]>select from_days(u32) from nums_all limit 1;
+----------------+
| from_days(u32) |
+----------------+
| 0090-01-24     |
+----------------+
1 row in set (0.13 sec)


mysql [ssb]>select from_days(u32) from nums_all limit 1;
+----------------+
| from_days(u32) |
+----------------+
| 0090-01-28     |
+----------------+
1 row in set (0.02 sec)
```

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: be/src/olap/parallel_scanner_builder.cpp [hunk 1]
```diff
@@ -199,9 +199,8 @@ Status ParallelScannerBuilder::_load() {
 std::shared_ptr<NewOlapScanner> ParallelScannerBuilder::_build_scanner(
         BaseTabletSPtr tablet, int64_t version, const std::vector<OlapScanRange*>& key_ranges,
         TabletReader::ReadSource&& read_source) {
-    NewOlapScanner::Params params {
-            _state,  _scanner_profile.get(), key_ranges,         std::move(tablet),
-            version, std::move(read_source), _limit_per_scanner, _is_preaggregation};
+    NewOlapScanner::Params params {_state,  _scanner_profile.get(), key_ranges, std::move(tablet),
+                                   version, std::move(read_source), _limit,     _is_preaggregation};
```
Comment (HappenLee) [inline be/src/olap/parallel_scanner_builder.cpp:203]: remove the useless space

**Review by HappenLee — APPROVED** [review]
LGTM

**Comment by zhangstar333** [comment]
run buildall

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**